api/v1beta1/azuremachine_default.go

```diff
@@ -188,15 +188,39 @@ func GetSubscriptionID(cli client.Client, clusterName string, namespace string,
 		err := cli.Get(ctx, key, ownerCluster)
 		if err != nil {
 			if i > maxAttempts {
-				return "", errors.Wrapf(err, "failed to find owner cluster for %s/%s", namespace, clusterName)
+				return "", "", errors.Wrapf(err, "failed to find owner cluster for %s/%s", namespace, clusterName)
 			}
 			time.Sleep(1 * time.Second)
 			continue
 		}
 		break
 	}
 
-	return ownerCluster.Spec.SubscriptionID, nil
+	return ownerCluster.Spec.InfrastructureRef.Name, ownerCluster.Spec.InfrastructureRef.Namespace, nil
+}
+
+// GetSubscriptionID returns the subscription ID for the AzureCluster given the cluster name and namespace.
+func GetSubscriptionID(cli client.Client, ownerAzureClusterName string, ownerAzureClusterNamespace string, maxAttempts int) (string, error) {
+	ctx := context.Background()
+
+	ownerAzureCluster := &AzureCluster{}
+	key := client.ObjectKey{
+		Namespace: ownerAzureClusterNamespace,
+		Name:      ownerAzureClusterName,
+	}
+	for i := 1; ; i++ {
+		err := cli.Get(ctx, key, ownerAzureCluster)
+		if err != nil {
+			if i > maxAttempts {
```

Comment:
nit: I don't think it really matters and this is copypasta from above, but it looks like this will actually do `maxAttempts+1` attempts.
```suggestion
			if i >= maxAttempts {
```

Reply:
Good catch!